core/src/test/java/org/elasticsearch/search/suggest/SuggestBuilderTests.java

```diff
@@ -39,27 +42,117 @@
+        suggesters = new Suggesters(new HashMap<>());
+        parseFieldMatcher = ParseFieldMatcher.STRICT;
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        namedWriteableRegistry = null;
+        queriesRegistry = null;
+        suggesters = null;
+        parseFieldMatcher = null;
+    }
 
     @Override
     protected NamedWriteableRegistry provideNamedWritableRegistry() {
-        NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry();
-        namedWriteableRegistry.registerPrototype(SuggestionBuilder.class, TermSuggestionBuilder.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SuggestionBuilder.class, PhraseSuggestionBuilder.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SuggestionBuilder.class, CompletionSuggestionBuilder.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SmoothingModel.class, Laplace.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SmoothingModel.class, LinearInterpolation.PROTOTYPE);
-        namedWriteableRegistry.registerPrototype(SmoothingModel.class, StupidBackoff.PROTOTYPE);
         return namedWriteableRegistry;
     }
 
     /**
+     * Test that valid JSON suggestion request passes.
+     */
+    public void testValidJsonRequestPayload() throws Exception {
+        final String field = RandomStrings.randomAsciiOfLength(getRandom(), 10).toLowerCase(Locale.ROOT);
+        String payload = "{\n" +
+                         "  \"valid-suggestion\" : {\n" +
+                         "    \"text\" : \"the amsterdma meetpu\",\n" +
+                         "    \"term\" : {\n" +
+                         "      \"field\" : \"" + field + "\"\n" +
+                         "    }\n" +
+                         "  }\n" +
+                         "}";
+        try {
+            final SuggestBuilder suggestBuilder = SuggestBuilder.fromXContent(newParseContext(payload), suggesters, true);
+            assertNotNull(suggestBuilder);
+        } catch (Exception e) {
+            fail("Parsing valid json should not have thrown exception: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that a malformed JSON suggestion request fails.
+     */
+    public void testMalformedJsonRequestPayload() throws Exception {
+        final String field = RandomStrings.randomAsciiOfLength(getRandom(), 10).toLowerCase(Locale.ROOT);
+        // {"bad-payload":{"prefix":"sug","completion":{"field":"ytnahgylcc","payload":[{"payload":"field"}]}}}
+        String payload = "{\n" +
+                         "  \"bad-payload\" : {\n" +
+                         "    \"text\" : \"the amsterdma meetpu\",\n" +
+                         "    \"term\" : {\n" +
+                         "      \"field\" : { \"" + field + "\" : \"bad-object\" }\n" +
+                         "    }\n" +
+                         "  }\n" +
+                         "}";
```

Comment:
@cbuescher I did change the test to use a string JSON representation instead of a builder, but the JSON itself was not the problem, it was the problem I highlighted in this comment: https://github.com/elastic/elasticsearch/pull/16873/files#r55274202